hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/sigs/utils/MiscCryptoUtils.java

```diff
@@ -18,24 +18,34 @@
 
 import static com.hedera.node.app.service.mono.utils.EntityIdUtils.EVM_ADDRESS_SIZE;
 
+import com.sun.jna.ptr.LongByReference;
+import java.nio.ByteBuffer;
 import java.util.Arrays;
 import org.bouncycastle.jcajce.provider.digest.Keccak;
-import org.bouncycastle.jce.ECNamedCurveTable;
-import org.bouncycastle.jce.spec.ECNamedCurveParameterSpec;
-import org.bouncycastle.math.ec.ECCurve;
+import org.hyperledger.besu.nativelib.secp256k1.LibSecp256k1;
 
 public class MiscCryptoUtils {
     private MiscCryptoUtils() {
         throw new UnsupportedOperationException("Utility Class");
     }
 
-    private static final ECNamedCurveParameterSpec secp256k1 = ECNamedCurveTable.getParameterSpec("secp256k1");
-    private static final ECCurve curveSecp256k1 = secp256k1.getCurve();
-
     public static byte[] keccak256DigestOf(final byte[] msg) {
         return new Keccak.Digest256().digest(msg);
     }
 
+    // Thread local caches to avoid allocating memory for each verification. They will leak memory for each thread used
+    // for verification but only just over 100 bytes to totally worth it.
+    private static final ThreadLocal<LibSecp256k1.secp256k1_pubkey> PUB_KEY_CACHE =
```

Comment:
Why not use a single `ThreadLocal` variable wrapping all the components? (`ThreadLocal.get()` is not cheap)

Reply:
Can do, I tested with and without in JMH and it was 10x gain and I stopped there and was happy LOL